pkg/transforms/argoapplication.go

```diff
@@ -0,0 +1,81 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package transforms
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ArgoApplicationResource ...
+type ArgoApplicationResource struct {
+	node Node
+}
+
+type ArgoApplication struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+	Spec              ArgoApplicationSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+}
+
+type ArgoApplicationSpec struct {
+	Source      ArgoApplicationSource      `json:"source" protobuf:"bytes,1,opt,name=source"`
+	Destination ArgoApplicationDestination `json:"destination" protobuf:"bytes,2,name=destination"`
+}
+
+type ArgoApplicationSource struct {
+	RepoURL        string `json:"repoURL" protobuf:"bytes,1,opt,name=repoURL"`
+	Path           string `json:"path,omitempty" protobuf:"bytes,2,opt,name=path"`
+	TargetRevision string `json:"targetRevision,omitempty" protobuf:"bytes,4,opt,name=targetRevision"`
+	Chart          string `json:"chart,omitempty" protobuf:"bytes,12,opt,name=chart"`
+}
+
+type ArgoApplicationDestination struct {
+	Server    string `json:"server,omitempty" protobuf:"bytes,1,opt,name=server"`
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,2,opt,name=namespace"`
+	Name      string `json:"name,omitempty" protobuf:"bytes,3,opt,name=name"`
+}
+
+// ArgoApplicationResourceBuilder ...
+func ArgoApplicationResourceBuilder(a *ArgoApplication) *ArgoApplicationResource {
+	node := transformCommon(a)
+	apiGroupVersion(a.TypeMeta, &node) // add kind, apigroup and version
+
+	// Extract the properties specific to this type
+	// Destination properties
+	if a.Spec.Destination.Name != "" {
+		node.Properties["destinationName"] = a.Spec.Destination.Name
+	}
+	if a.Spec.Destination.Namespace != "" {
+		node.Properties["destinationNamespace"] = a.Spec.Destination.Namespace
+	}
+	if a.Spec.Destination.Server != "" {
+		node.Properties["destinationServer"] = a.Spec.Destination.Server
+	}
```

Comment:
This pattern may cause problems if any of these properties is removed during an update.  Let's send the empty value, this guarantees that the existing value is removed.

_Also sending the empty value keeps the data schema always the same, which might help us in the future when we replace Redisgraph._